docs/product-user-guide/src/main/asciidoc/drl-rules-THEN-proc.adoc

```diff
@@ -0,0 +1,102 @@
+[id='drl-rules-THEN-proc']
+= Adding THEN actions in DRL rules
+
+The `then` part of the rule contains the actions to be performed when the conditional part of the rule has been met. For example, when a loan applicant is under 21 years old, the `then` action of an "Underage" rule would be `setApproved( false )`, declining the loan because the applicant is under age. Actions execute consequences based on the rule conditions and on available data objects in the package.
+
+.Prerequisites
+* The `package` is defined at the top of the DRL file. This should have been done for you when you created the file.
+* The `import` list of data objects used in the rule is defined below the `package` line of the DRL file. Data objects can be from this package or from another package in {CENTRAL}.
+* The `rule` name is defined in the format `rule "name"` below the `package`, `import`, and other lines that apply to the entire DRL file. The same rule name cannot be used more than once in the same package. Optional rule attributes (such as `salience` or `no-loop`) that define rule behavior are below the rule name, before the `when` section.
+
+.Procedure
+. In the DRL designer, enter `then` after the `when` section of the rule to begin adding action statements.
+. Enter one or more actions to be executed on fact patterns based on the conditions for the rule.
++
+Below are some of the keyword options for defining DRL actions:
++
+* `and`: Use this to group action components into a logical conjunction. Infix and prefix `and` are supported. By default, all listed conditions or actions are combined with `and` when no conjunction is specified.
++
+[source,java]
+----
+application.setApproved ( false ) and application.setExplanation( "has been bankrupt" );
+
+application.setApproved ( false );
+and application.setExplanation( "has been bankrupt" );
+
+application.setApproved ( false );
+application.setExplanation( "has been bankrupt" );
+
+// All of the above are the same.
+----
++
+* `or`: Use this to group action components into a logical disjunction. Infix and prefix `or` are supported.
++
+[source,java]
+----
+( Bankruptcy( amountOwed == 100000 ) or IncomeSource( amount == 20000 ) )
+
+Bankruptcy( amountOwed == 100000 )
+or IncomeSource( amount == 20000 )
+----
++
+* `set`: Use this to set the value of a field.
++
+[source,java]
+----
+application.setApproved ( false );
+application.setExplanation( "has been bankrupt" );
+----
++
+* `modify`: Use this to specify fields to be modified for a fact and to notify the decision engine of the change.
++
+[source,java]
+----
+modify( LoanApplication ) {
+        setAmount( 100 )
+}
+----
++
+* `update`: Use this to specify fields and the entire related fact to be modified and to notify the decision engine of the change. After a fact has changed, you must call `update` before changing another fact, or indexing problems can occur within the decision engine. The `modify` keyword avoids this problem.
```

Comment:
@sterobin It's not really that "indexing problems can occur" but other rules that may match the updated fact will not be notified of the change and hence given an opportunity to re-evaluate themselves against the updated values.